### ISSUE TITLE
Extract title and content from Safari

### ISSUE THREAD

**Issue by aniqueta** [jandamm/wallabag-macos#23]
@jandamm Can this extension adopt the functionality from the iOS app, per https://github.com/wallabag/ios-app/pull/420? Thanks.

**Comment by jandamm** [comment]
Hey @aniqueta, it's easy to get the title of the website since it is included in the API I'm using.
The iOS app is using a different API, I'd need to investigate how I can get the content of the current page.

**Comment by aniqueta** [comment]
Thanks. This would be quite useful for increasing difficulty to get content for paywalled content on the server side.

**Comment by aniqueta** [comment]
Just wanted to follow and see if this feature is planned? More and more sites are not saving from server side and thus a browser based way would be great. Thanks.

**Comment by jandamm** [comment]
Thanks for following up @aniqueta.
I planned on implementing this feature.

Unfortunately Apple halted any updates to the app: https://github.com/jandamm/wallabag-macos/commit/73f90cb4c61bfdafce98ecee3a69bbddf0a5dfe9

I've written a mail to @nicosomb if it's okay for the app to be in the AppStore with `Wallabag` in the name and if it's okay that I could get some form of allowance to resubmit the app. Either by listing the extension on wallabag.org or in another form.

Until I cannot get Wallabag 2025.4 approved there is no sense to implement this feature 😔

**Comment by nicosomb** [comment]
@jandamm did I reply? (or I forgot to do it) 🫣

**Comment by jandamm** [comment]
No you didn't 😅

**Comment by nicosomb** [comment]
I can't find it by searching with your firstname and lastname. Could you send it again?

**Comment by jandamm** [comment]
@nicosomb I just sent it again (at nicolas@wallabag.it)

**Comment by jandamm** [comment]
Thanks for the reply, @nicosomb.
I've created a PR to add this extension on the Website: https://github.com/wallabag/website/pull/97

**Comment by afonsoguerra** [comment]
Now that updates are working, is it possible to implement this feature? Happy to beta test it if it is any help.

**Comment by jandamm** [comment]
Yes it would be possible to implement this feature now. Unfortunately I don't have any capacity to implement it in the near future.
I would however create a beta version if someone would create a PR. Otherwise I'll have to see when I have time to do any coding.

There are already some parts implemented through #26. At least injecting a script and calling it from the native code is implemented there.

**Comment by afonsoguerra** [comment]
Is there still a plan to implement this in the future?

**Comment by aniqueta** [comment]
I've been using the Wallabag iOS app's share extensions on macOS when the extension doesn't work. Kludgey but works for full text some of the time.

**Comment by afonsoguerra** [comment]
The problem is there is a bunch of websites that don't like to be parsed by wallabag... The workaround I have is to use the rather nice obsidian-web-clipper extension, and then hack the resulting .md file back into wallabag... It works but it's a pain

**Comment by jandamm** [comment]
Yes, I still plan to implement this feature ~~but don't have the capacity to do so~~ (turned out to be much faster than I previously thought).

**Comment by jandamm** [comment]
I've drafted something which should enable this functionality.
Right now there is only one button to enable or disable this for all websites.
This should also work for translated websites (#24)

Please write me a mail to wallamac@jandamm.de if you want to be invited to the beta test.

@aniqueta, @afonsoguerra, @alex-chris

**Comment by jandamm** [comment]
The beta is ready to test.
You should've gotten an invitation by now. Otherwise contact me via Mail again.

**Comment by afonsoguerra** [comment]
Thanks for the beta. I can see the button in the config and the different options in the context menu, but when I try to get a 'dificult' website, for instance https://www.nature.com/articles/d41586-026-00239-y it still results in empty content.

**Comment by aniqueta** [comment]
Just tested with a bunch of NY Times paywalled articles while logged in, and likewise, also getting the prior behavior of "wallabag can't retrieve contents for this article."

**Comment by jandamm** [comment]
Okay, thanks for testing. I need to investigate this.

**Comment by jandamm** [comment]
@afonsoguerra @aniqueta I think I fixed the issue now. There is a new beta on it's way 👍

**Comment by afonsoguerra** [comment]
Awesome stuff! I was ping'ed by my computer that something new was available and I can happily confirm that it works much better! 

Update to say that actually all images turn up in the server, so it was just a sync thing, it all goes perfectly to the reader so I think this is fixed for me.

Many thanks for spending your time on this! It will undoubtedly save me a lot of time in the future!

**Comment by jandamm** [comment]
I've submitted this version to apple for review. I hope it'll be live in the AppStore soon. 🎉🚀

**Comment by jandamm** [comment]
Hopefully it's live soon. Apple took quite some time to start reviewing the app.
Then they noticed that the test credentials were expired.
I re-submitted with new credentials and now they rejected again because they still used the old credentials 😓🫣

**Comment by jandamm** [comment]
Apple is being funny.
Now they rejected because they couldn't find the saved article in the app 🙊 

<img width="1132" height="234" alt="Rejection" src="https://github.com/user-attachments/assets/474b67e0-23b6-403e-ba24-bc6632f795e7" />

**Comment by jandamm** [comment]
Apple did finally approve.

Should be live in the AppStore soon.

**Comment by aniqueta** [comment]
Thank you. Sent through a tip. :)

**Comment by jandamm** [comment]
Thank you very much ❤️
